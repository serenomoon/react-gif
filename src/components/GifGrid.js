import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem'
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {


  const {data:images, loading} = useFetchGifs(category);

  


  return (
    <>
      <h2 className='animate__animated animate__fadeIn'>{category}</h2>

      {loading && <p className='animate__animated animate__flash'>loading...</p>}

      <div className='card-grid'>

            {
              images.map(img => ( 
                <GifGridItem 
                    key={img.id}
                    {...img}
                  />
              ))
            }

      </div>
    </>
  );
};

GifGrid.propTypes = {
  category : PropTypes.string.isRequired
}
