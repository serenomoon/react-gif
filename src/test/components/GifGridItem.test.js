import React from 'react'
import { shallow } from "enzyme";
import '@testing-library/jest-dom';
import toJson from 'enzyme-to-json';
import { GifGridItem } from "../../components/GifGridItem"

describe('<GifGridItem.js /> Testing', () => {
    
    let wrapper = shallow(<GifGridItem />)

    test('Debe mostrar el snapshot correctamente', () => {
        
        expect(wrapper).toMatchSnapshot();

    });

});