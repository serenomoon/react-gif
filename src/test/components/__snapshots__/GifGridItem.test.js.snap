// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGridItem.js /> Testing Debe mostrar el snapshot correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img />
  <p />
</div>
`;
